src/parse.c

```diff
@@ -2609,3 +2617,70 @@ netplan_clear_netdefs()
 	}
     return n;
 }
+
+void
+process_input_file(const char* f)
+{
+    GError* error = NULL;
+
+    g_debug("Processing input file %s..", f);
+    if (!netplan_parse_yaml(f, &error)) {
+        g_fprintf(stderr, "%s\n", error->message);
+        exit(1);
+    }
+}
+
+gboolean
+process_yaml_hierarchy(const char* rootdir)
+{
+    glob_t gl;
+    /* Files with asciibetically higher names override/append settings from
+     * earlier ones (in all config dirs); files in /run/netplan/
+     * shadow files in /etc/netplan/ which shadow files in /lib/netplan/.
+     * To do that, we put all found files in a hash table, then sort it by
+     * file name, and add the entries from /run after the ones from /etc
+     * and those after the ones from /lib. */
+    if (find_yaml_glob(rootdir, &gl) != 0)
+        return FALSE; // LCOV_EXCL_LINE
+    /* keys are strdup()ed, free them; values point into the glob_t, don't free them */
+    g_autoptr(GHashTable) configs = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, NULL);
+    g_autoptr(GList) config_keys = NULL;
+
+    for (size_t i = 0; i < gl.gl_pathc; ++i)
+        g_hash_table_insert(configs, g_path_get_basename(gl.gl_pathv[i]), gl.gl_pathv[i]);
+
+    config_keys = g_list_sort(g_hash_table_get_keys(configs), (GCompareFunc) strcmp);
+
+    for (GList* i = config_keys; i != NULL; i = i->next)
+        process_input_file(g_hash_table_lookup(configs, i->data));
+    return TRUE;
+}
+
+gchar*
+netplan_get_id_from_filename(const char* filename, const char* ssid)
```

Comment:
This is entirely fine, this function makes sense. One thing I'd like us to consider (now or at some later point, really) is to maybe change the name of this function. Since this is a NetworkManager-backend specific function, only useful whenever NM is in use. The `netplan_get_id_from_filename()` name seems to give off a vibe that it's for some general ID stored in the filename.
So maybe worth considering: changing the name to something like `netplan_get_nm_id_from_filename()` (or maybe some better placement of the `nm` part) and maybe moving this to `util.h` instead?

Reply:
Yes... I was thinking that this function could be extended to also extract names from networkd filenames. But I agree that we don't have (nor need) this functionality currently, so we shouldn't name the function like this. We can still wrap around a more generic function(-name), once it is implemented in a more generic way.

I renamed it to `netplan_get_id_from_nm_filename` (as it only works for NetworkManager connection profile filenames, generated by netplan) and moved it into util.c/.h

Merging (and rebasing #182 on top of it).